Make it clearer that `before_send` can drop events (#8300)
getsentry/sentry-docs: src/platforms/common/configuration/filtering.mdx

### Using <PlatformIdentifier name="before-send" />

> All Sentry SDKs support the <PlatformIdentifier name="before-send" /> callback method. <PlatformIdentifier name="before-send" /> is called immediately before the event is sent to the server, so it’s the final place where you can edit its data or decide to not send it at all. It receives the event object as a parameter, so you can use that to modify the event’s data or drop it completely (by returning `null`) based on custom logic and the data available on the event.

Reviewer: ```suggestion
All Sentry SDKs support the <PlatformIdentifier name="before-send" /> callback method.  Because it's called immediately before the event is sent to the server, this is your last chance to decide not to send data or to edit it. <PlatformIdentifier name="before-send" /> receives the event object as a parameter, which you can use to either modify the event’s data or drop it completely by returning `null`, based on custom logic and the data available on the event.
```
